internal/typeinfo/arginfo.go

```diff
@@ -87,7 +87,16 @@ func (argInfo ArgInfo) AllStructOutputs(typeName string) ([]Output, []string, er
 	}
 	si, ok := arg.(*structInfo)
 	if !ok {
-		return nil, nil, fmt.Errorf("columns must be specified for non-struct type")
+		errStr := ""
+		switch arg.typ().Kind() {
+		case reflect.Map:
+			errStr = "cannot use %s with asterisk unless columns are specified"
+		case reflect.Slice:
+			errStr = "cannot use %s with asterisk"
+		default:
+			errStr = "internal error: unknown invalid arg type %s"
```

Comment:
[Nit]: What about a return instead or creating the error directly using `fmt.Errorf`? It is a little bit weird to do a switch on `arg.typ().Kind()` just to add it later.

Reply:
Fair enough, changed it to just return. I wrote it like this because the line length was annoying me but its not actually that bad.